app/main/views/agencies.py

```diff
@@ -24,3 +24,24 @@ def update_agency(agency_id):
     agency_business.update(agency_id, updated, update_details.get('updated_by'))
     agency = agency_business.get_agency(agency_id)
     return jsonify(agency=agency)
+
+
+@main.route('/admin/create_agency', methods=['POST'])
+def create_agency():
+    json_payload = get_json_from_request()
+    data = json_payload.get('data', None)
+
+    if (data['state'] == ''):
+        data['state'] = None
+    try:
+        agency = agency_business.insert_agency(data['name'],
+                                               data['domains'],
+                                               data['bodyType'],
+                                               data['category'],
+                                               data['state'],
+                                               data['reports'],
+                                               data['whitelisted'],
+                                               data['must_join_team'])
+        return jsonify({'id': str(agency.id)})
+    except Exception:
+        return 'Duplicate domain name', 400
```

Comment:
I think you should be more specific with exceptions.  What if a different error occurs?  This message may not apply.